src/Repository/CommitteeMembershipRepository.php

```diff
@@ -361,61 +303,76 @@ function (CommitteeMembership $membership) {
      */
     public function findCommitteesUuidByHostFirstName(string $firstName): array
     {
-        $qb = $this->createQueryBuilder('cm');
-
-        $query = $qb
-            ->select('cm.committeeUuid')
-            ->leftJoin('cm.adherent', 'a')
-            ->where('LOWER(a.firstName) LIKE :firstName')
-            ->andWhere($qb->expr()->in('cm.privilege', CommitteeMembership::getHostPrivileges()))
-            ->setParameter('firstName', '%'.strtolower($firstName).'%')
-            ->getQuery()
-        ;
-
-        return array_map(function (UuidInterface $uuid) {
-            return $uuid->toString();
-        }, array_column($query->getArrayResult(), 'committeeUuid'));
+        return $this->findCommitteesUuid([
+            'firstName' => $firstName,
+            'privileges' => CommitteeMembership::getHostPrivileges(),
+        ]);
     }
 
     /**
      * @return string[]
      */
     public function findCommitteesUuidByHostLastName(string $lastName): array
     {
-        $qb = $this->createQueryBuilder('cm');
-
-        $query = $qb
-            ->select('cm.committeeUuid')
-            ->leftJoin('cm.adherent', 'a')
-            ->where('LOWER(a.lastName) LIKE :lastName')
-            ->andWhere($qb->expr()->in('cm.privilege', CommitteeMembership::getHostPrivileges()))
-            ->setParameter('lastName', '%'.strtolower($lastName).'%')
-            ->getQuery()
-        ;
-
-        return array_map(function (UuidInterface $uuid) {
-            return $uuid->toString();
-        }, array_column($query->getArrayResult(), 'committeeUuid'));
+        return $this->findCommitteesUuid([
+            'lastName' => $lastName,
+            'privileges' => CommitteeMembership::getHostPrivileges(),
+        ]);
     }
 
     /**
      * @return string[]
      */
     public function findCommitteesUuidByHostEmailAddress(string $emailAddress): array
     {
-        $qb = $this->createQueryBuilder('cm');
+        return $this->findCommitteesUuid([
+            'emailAddress' => $emailAddress,
+            'privileges' => CommitteeMembership::getHostPrivileges(),
+        ]);
+    }
 
-        $query = $qb
-            ->select('cm.committeeUuid')
-            ->leftJoin('cm.adherent', 'a')
-            ->where('LOWER(a.emailAddress) LIKE :emailAddress')
-            ->andWhere($qb->expr()->in('cm.privilege', CommitteeMembership::getHostPrivileges()))
-            ->setParameter('emailAddress', '%'.strtolower($emailAddress).'%')
-            ->getQuery()
+    public function findCommitteesUuid(array $criteria): array
```

Comment:
IMHO it's better to have some private methods that can build a base line querybuilder than trying to build something generic with an array of `$criteria`. 

At the beginning that sort of method are small and simple and they end big and cumbersome for no good reasons than trying to make generic code.